src/functions.sh

```diff
@@ -37,11 +40,22 @@ has_shebang () {
   local file="$1"
 
   if IFS= read -r line < "./${file}" ; then
-    case $line in
-      "#!/bin/bash") return 0;;
-      "#!/bin/sh") return 0;;
-      *) return 2
-    esac
+    local shebang_regexp="\s*\#\!\s*(\/usr(\/local)?)?\/bin\/(env\s+)?interpreter\s*"
```

Comment:
```suggestion
    local shebang_regexp="^\s*(\#|\!)?\s*(\#\!)\s*(\/usr(\/local)?)?\/bin\/(env\s+)?interpreter\b"
```

Reply:
I would probably use something like this:

```shell
^\s*((\#|\!)|(\#\s*\!)|(\!\s*\#))\s*(\/usr(\/local)?)?\/bin\/(env\s+)?interpreter\b
```

Reply:
Sure. 😃